src/zasm/src/core/filestream.cpp

```diff
@@ -88,9 +88,9 @@ namespace zasm
 
         FILE* fp = nullptr;
 #ifdef _WIN32
-        _wfopen_s(&fp, path.wstring().c_str(), mode == StreamMode::Read ? L"rb" : L"wb");
+        _wfopen_s(&fp, path.wstring().c_str(), mode == StreamMode::Read ? L"rb" : mode == StreamMode::Write ? L"wb" : L"r+b");
```

Comment:
I'm not a fan of this, create a new static function that has `static const char* getInternalMode(StreamMode mode)`

Reply:
> I'm not a fan of this, create a new static function that has `static const char* getInternalMode(StreamMode mode)`

Ok I did but it seems that _wfopen_s requires wchar_t with L literal and your function return type works only for fopen .Should I create another function ?